draft-ietf-tokbind-https-06.xml

```diff
@@ -937,6 +960,33 @@ contexts. Other approaches are possible, but are outside the scope of this speci
           mechanisms.
         </t>
       </section>
+      <section title="Correlation" anchor="Correlation">
+        <t>
+	  Having Token Bindings using other than the provided Token Binding type for the current TLS connection
+	  gives communicating parties information about additional TLS connections of the application.
+	  The Token Binding IDs in these additional Token Bindings
+	  can serve as correlation handles for the endpoints of those connections.
+	  If the application is already aware of these other connections to these endpoints,
+	  then no additional information is being exposed.
+	  For instance, if in a redirect-based federation protocol,
+	  the Identity Provider and Relying Party already have URLs for one another,
+	  also having Token Binding information for these connections
+	  does not provide additional correlation information.
+	  However, if not, by providing Token Bindings,
+	  then there is potential to expose additional information
+	  that could be used to correlate the endpoints that different parties are communicating with.
+	  In these cases, a privacy analysis of correlations enabled
+	  and the potential privacy impacts of those correlations should be performed
+	  as part of the decision whether to utilize token binding in those cases.
+	</t>
+	<t>
+	  Care should be taken to ensure that unrelated applications cannot obtain information
+	  about each other's Token Bindings.
+	  For instance, a Token Binding implementation shared between multiple applications on a machine
+	  should prevent unrelated applications from seeing each other's Token Binding information.
+	  Application isolation and key segregation should be used to prevent this.
+	</t>
```

Comment:
wordsmith suggestions on the above two paragraphs...

<t>
An application's communicating parties, that receive Token Bindings for TLS connections other than their own, obtain information about the application's other TLS connections. The Token Binding IDs in these other Token Bindings can serve as correlation handles for the endpoints of the other connections.
If the application is already aware of these other connections, then no additional information is being exposed. For instance, if in a redirect-based federation protocol, the Identity Provider and Relying Party already have URLs for one another, also having Token Binding information for these connections does not provide additional correlation information.
If not, then by providing the other Token Bindings there is potential to expose additional information that can be used to correlate the endpoints the other parties are communicating with.
In such cases, a privacy analysis of enabled correlations and their potential privacy impacts should be performed as part of the application design decisions of how, and whether, to utilize Token Bindings.
</t>
<t>
Also, care should be taken to ensure that unrelated applications cannot obtain information about each other's Token Bindings. For instance, a Token Binding implementation shared between multiple applications on a given system should prevent unrelated applications from obtaining each other's Token Binding information. This may be accomplished by using  techniques such as application isolation and key segregation, depending upon system capabilities.
</t>

HTH,

=JeffH